docs/project-model/project-json-reference.md

```diff
@@ -0,0 +1,494 @@
+* [compilerName](#compilerName)
+* [testRunner](#testRunner)
+* [authors](#authors)
+* [owners](#owners)
+* [tags](#tags)
+* [language](#language)
+* [releaseNotes](#releaseNotes)
+* [requireLicenseAcceptance](#requireLicenseAcceptance)
+* [embedInteropTypes](#embedInteropTypes)
+* [compile](#compile)
+* [content](#content)
+* [resource](#resource)
+* [preprocess](#preprocess)
+* [publishExclude](#publishExclude)
+* [shared](#shared)
+* [namedResource](#namedResource)
+* [packInclude](#packInclude)
+* [exclude](#exclude)
+* [contentBuiltIn](#contentBuiltIn)
+* [compileBuiltIn](#compileBuiltIn)
+* [resourceBuiltIn](#resourceBuiltIn)
+* [excludeBuiltIn](#excludeBuiltIn)
+* [dependencies](#dependencies)
+* [tools](#tools)
+* [commands](#commands)
+* [scripts](#scripts)
+* [compilationOptions](#compilationOptions)
+    * [define](#define)
+    * [nowarn](#nowarn)
+    * [additionalArguments](#additionalArguments)
+    * [languageVersion](#languageVersion)
+    * [allowUnsafe](#allowUnsafe)
+    * [platform](#platform)
+    * [warningsAsErrors](#warningsAsErrors)
+    * [optimize](#optimize)
+    * [keyFile](#keyFile)
+    * [delaySign](#delaySign)
+    * [publicSign](#publicSign)
+    * [emitEntryPoint](#emitEntryPoint)
+    * [xmlDoc](#xmlDoc)
+    * [preserveCompilationContext](#preserveCompilationContext)
+* [analyzerOptions](#analyzerOptions)
+    * [languageId](#languageId)
+* [configurations](#configurations)
+* [frameworks](#frameworks)
+    * [dependencies](#dependencies)
+    * [frameworkAssemblies](#frameworkAssemblies)
+    * [wrappedProject](#wrappedProject)
+    * [imports](#imports)
+    * [bin](#bin)
+        * [assembly](#assembly)
+
+<a name="name"></a>
+## name
+Type: String
+
+The name of the project, this is also the name of the (nuget) package the project will produce.
+
+<a name="version"></a>
+## version
+Type: String
+
+The semver version of the project, also used for the nuget package.
+
+<a name="description"></a>
+## description
+Type: String
+
+A longer decsiption of the project.
+
+<a name="summary"></a>
+## summary
+Type: String
+
+A short description of the project.
+
+<a name="copyright"></a>
+## copyright
+Type: String
+
+The copyright information for the project.
+
+<a name="title"></a>
+## title
+Type: String
+
+The friendly name of the project, can contain spaces and special characters not allowed in the name.
+
+<a name="entryPoint"></a>
+## entryPoint
+Type: String
+
+The entrypoint method for the project, typically Main().
+
+<a name="projectUrl"></a>
+## projectUrl
+Type: String
+
+The url for the homepage of the project.
+
+<a name="licenseUrl"></a>
+## licenseUrl
+Type: String
+
+The url for the licence the project uses.
+
+<a name="iconUrl"></a>
+## iconUrl
+Type: String
+
+The url for an icon that will be used in various places such as the package explorer.
+
+<a name="compilerName"></a>
+## compilerName
+Type: String
+
+The name of the compiler used for this project, for c# this will be "csc" for example.
```

Comment:
Default behavior should be specified here.  We should also specify the strings that represent valid compilers.  Currently that's "csc" and "fsc".  We'll have to define what those are as well.